pkg/alertmanager/multitenant.go

```diff
@@ -94,6 +94,11 @@ type MultitenantAlertmanagerConfig struct {
 	Peers                flagext.StringSlice `yaml:"peers"`
 	PeerTimeout          time.Duration       `yaml:"peer_timeout"`
 
+	HABindAddr      string                 `yaml:"ha_bind_address"`
+	HAAdvertiseAddr string                 `yaml:"ha_advertise_address"`
+	HAPeers         flagext.StringSliceCSV `yaml:"ha_initial_peers"`
+	HAPeerTimeout   time.Duration          `yaml:"ha_peer_timeout"`
```

Comment:
I am not too sure if `ha` is an improvement. I would prefer staying with `cluster` as alertmanager upstream is using:

```
      --cluster.listen-address="0.0.0.0:9094"
                                 Listen address for cluster. Set to empty string to disable HA mode.
      --cluster.advertise-address=CLUSTER.ADVERTISE-ADDRESS
                                 Explicit address to advertise in cluster.
      --cluster.peer=CLUSTER.PEER ...
                                 Initial peers (may be repeated).
      --cluster.peer-timeout=15s
                                 Time to wait between peers to send notifications.
      --cluster.gossip-interval=200ms
                                 Interval between sending gossip messages. By lowering this value (more frequent) gossip messages are propagated across the cluster more quickly at the expense of increased bandwidth.
      --cluster.pushpull-interval=1m0s
                                 Interval for gossip state syncs. Setting this interval lower (more frequent) will increase convergence speeds across larger clusters at the expense of increased bandwidth usage.
      --cluster.tcp-timeout=10s  Timeout for establishing a stream connection with a remote node for a full state sync, and for stream read and write operations.
      --cluster.probe-timeout=500ms
                                 Timeout to wait for an ack from a probed node before assuming it is unhealthy. This should be set to 99-percentile of RTT (round-trip time) on your network.
      --cluster.probe-interval=1s
                                 Interval between random node probes. Setting this lower (more frequent) will cause the cluster to detect failed nodes more quickly at the expense of increased bandwidth usage.
      --cluster.settle-timeout=1m0s
                                 Maximum time to wait for cluster connections to settle before evaluating notifications.
      --cluster.reconnect-interval=10s
                                 Interval between attempting to reconnect to lost peers.
      --cluster.reconnect-timeout=6h0m0s
```
Addtiionally I would also rather have a separate `ClusterConfig struct` so that the flag name separation with `.` is consistent with the YAML structure.

Also it might make sense to align the naming of parameters with our memberlist flags from here: https://github.com/cortexproject/cortex/blob/77c73aecaad5f81cba3fc4dd2bf5d4b501cb776c/pkg/ring/kv/memberlist/memberlist_client.go#L168-L186